src/Codeception/Subscriber/Dependencies.php

```diff
@@ -28,7 +28,8 @@ public function testStart(TestEvent $event)
 
         $testSignatures = $test->getDependencies();
         foreach ($testSignatures as $signature) {
-            if (!in_array($signature, $this->successfulTests)) {
+            $matches = preg_grep('/'.addslashes($signature).'(?::\d+)?/', $this->successfulTests);
```

Comment:
Please use preg_quote instead of addslashes and put spaces on both sides of concatenation operator.
Also I think that there is no guarantee that iteration is numeric,
dataProvider function can return named examples.
```php
return [
  'a' => ['param1', 'param2'],
  'b' => ['param1', 'param2'],
];
```